openedx/core/djangoapps/programs/utils.py

```diff
@@ -383,9 +383,6 @@ def _attach_run_mode_course_image_url(self, run_mode):
     def _attach_run_mode_course_url(self, run_mode):
         run_mode['course_url'] = reverse('course_root', args=[self.course_key])
 
-    def _attach_run_mode_end_date(self, run_mode):
```

Comment:
Please don't remove this, it is actively used. You didn't find it when searching because it's called using getattr. https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/programs/utils.py#L363